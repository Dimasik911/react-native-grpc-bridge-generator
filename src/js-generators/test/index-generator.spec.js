/* eslint-env jest */

import generator, { template } from '../index-generator'

describe('template', () => {
  it('creates template', () => {
    const rsp = template('ServiceName123')

    expect(rsp).toMatchSnapshot()
  })
})

describe('generator', () => {
  it('creates the out put with the correct service name', () => {
    let schema = {
      services: [{
        name: 'GrpcService',
        methods: [{
          name: 'ReverseGeocode',
          input_type: 'ReverseGeocodeRequest',
          output_type: 'Location',
          client_streaming: false,
          server_streaming: false,
          options: {}
        }]
      }]
    }
    const rsp = generator(schema)

    expect(rsp).toMatchSnapshot()
  })
})
