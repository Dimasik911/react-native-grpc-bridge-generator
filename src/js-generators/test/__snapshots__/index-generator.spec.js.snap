// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generator creates the out put with the correct service name 1`] = `
"// @flow

import * as actions from 'GrpcService-actions'
import * as actionTypes from 'GrpcService-action-types'
import * as apiClient from 'GrpcService-api-client'

const GrpcService = {
  actions,
  actionTypes,
  apiClient
}

module.exports = GrpcService"
`;

exports[`template creates template 1`] = `
"// @flow

import * as actions from 'ServiceName123-actions'
import * as actionTypes from 'ServiceName123-action-types'
import * as apiClient from 'ServiceName123-api-client'

const ServiceName123 = {
  actions,
  actionTypes,
  apiClient
}

module.exports = ServiceName123"
`;
